Test to verify write operation should fail from secondary site in rbd mirroring

sunilangadi2 opened the pull request:
# Description
Automated below test case
https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-9503

As discussed with team since it's just one negative test to validate, hence accommodate that test in existing 
`tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py` as it covers both validation from `journal` and `snapshot` based mirroring.

Test flow:

- On the remote/secondary cluster, try using the mirrored image for writing.
- command should error out saying `Read only file system`

Test Result:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-BBJH8C



Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

manasagowri commented on tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py:
```diff
@@ -5,6 +5,41 @@
+        0 - if test case pass
+        1 - it test case fails
+    """
+    out, err = self.exec_cmd(
```
Please use the benchwrite method present in rbd_mirror_utils for this instead of executing your own command again.
Let's reuse existing common methods.

manasagowri replied:
If you need that method to return both out and err modify that method to return the same as an optional return whenever output is requested.

You can take kw["output"] = True as input to benchwrite method and if it is set to true then you return both out and err, something similar to exec_cmd method in rbd_utils.

sunilangadi2 replied:
Initially me to thought the same way and tried but https://github.com/red-hat-storage/cephci/blob/master/tests/rbd_mirror/rbd_mirror_utils.py#L62
as bench write is executing through `long running` it just return `out` no `error`
Hence used the direct rbd bench.

manasagowri commented on tests/rbd_mirror/test_rbd_mirror_journal_to_snap.py:
```diff
@@ -5,6 +5,41 @@
+        return 0
+
+    log.error(
+        f"In {mirrormode} based able to write data from secondary, even mirrored image got locked from primary"
```
Could you please change this to, 
**Able to write data into secondary image for {mirrormode} based mirroring. This is invalid since image should be locked at primary**

I felt this seems more easily understandable. Please let me know your thoughts.

sunilangadi2 replied:
Done, Thanks

sunilangadi2 commented:
@manasagowri @rkhambad Addressed all review comments, Covered EC pool coverage as well,
Please review it.
Test results:
[Passed] http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-DMXXLM

sunilangadi2 commented:
Since this PR started with Redhat laptop author got mismatched with new laptop,
Due to that commits get conflicted
Raised New PR for the same with single comit
https://github.com/red-hat-storage/cephci/pull/2438

Marking this PR as closed.
